atomics/include/desul/atomics/Compare_Exchange_SYCL.hpp

```diff
@@ -20,86 +20,76 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #else
 #include <CL/sycl.hpp>
 #endif
-// clang-format on
-
-#ifdef DESUL_HAVE_SYCL_ATOMICS
 
 namespace desul {
+namespace Impl {
 
-template <class MemoryOrder, class MemoryScope>
-inline void atomic_thread_fence(MemoryOrder, MemoryScope) {
-  sycl::atomic_fence(
-      Impl::DesulToSYCLMemoryOrder<MemoryOrder, /*extended namespace*/ false>::value,
-      Impl::DesulToSYCLMemoryScope<MemoryScope, /*extended namespace*/ false>::value);
-}
-
-template <typename T, class MemoryOrder, class MemoryScope>
-typename std::enable_if<sizeof(T) == 4, T>::type atomic_compare_exchange(
+template <class T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<sizeof(T) == 4, T> host_atomic_compare_exchange(
     T* const dest, T compare, T value, MemoryOrder, MemoryScope) {
   static_assert(sizeof(unsigned int) == 4,
                 "this function assumes an unsigned int is 32-bit");
-  Impl::sycl_atomic_ref<unsigned int, MemoryOrder, MemoryScope> dest_ref(
+  sycl_atomic_ref<unsigned int, MemoryOrder, MemoryScope> dest_ref(
       *reinterpret_cast<unsigned int*>(dest));
   dest_ref.compare_exchange_strong(*reinterpret_cast<unsigned int*>(&compare),
                                    *reinterpret_cast<unsigned int*>(&value));
   return compare;
 }
-template <typename T, class MemoryOrder, class MemoryScope>
-typename std::enable_if<sizeof(T) == 8, T>::type atomic_compare_exchange(
+
+template <class T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<sizeof(T) == 8, T> host_atomic_compare_exchange(
     T* const dest, T compare, T value, MemoryOrder, MemoryScope) {
   static_assert(sizeof(unsigned long long int) == 8,
                 "this function assumes an unsigned long long is 64-bit");
-  Impl::sycl_atomic_ref<unsigned long long int, MemoryOrder, MemoryScope> dest_ref(
+  sycl_atomic_ref<unsigned long long int, MemoryOrder, MemoryScope> dest_ref(
       *reinterpret_cast<unsigned long long int*>(dest));
   dest_ref.compare_exchange_strong(*reinterpret_cast<unsigned long long int*>(&compare),
                                    *reinterpret_cast<unsigned long long int*>(&value));
   return compare;
 }
 
-template <typename T, class MemoryOrder, class MemoryScope>
-typename std::enable_if<sizeof(T) == 4, T>::type atomic_exchange(T* const dest,
-                                                                 T value,
-                                                                 MemoryOrder,
-                                                                 MemoryScope) {
+template <class T, class MemoryOrder, class MemoryScope>
+std::enable_if_t<sizeof(T) == 4, T> host_atomic_exchange(T* const dest,
```

Comment:
Why is this called `host_atomic_exchange`? Before this pull request, we were only using the `SYCL` atomics on the device anyway.

Reply:
but the previous overload (atomic_exchange) was available on both host and device wasn't it?

Reply:
Yes, but there were some problems using it on the host.

I haven't tried to understand the strategy for the renaming of functions and it might be correct to call it `host_atomic_exchange`. I'm just asking for clarification.

Reply:
We were going through the SYCL atomics both on the host and the device side.  I updated so we only use them on the device side.